Makefile

```diff
@@ -40,14 +40,30 @@ ifeq (${CDC_ENABLE_VENDOR}, 1)
 GOVENDORFLAG := -mod=vendor
 endif
 
-GOBUILD  := CGO_ENABLED=0 $(GO) build $(BUILD_FLAG) -trimpath $(GOVENDORFLAG)
-GOBUILDNOVENDOR  := CGO_ENABLED=0 $(GO) build $(BUILD_FLAG) -trimpath
-GOTEST   := CGO_ENABLED=1 $(GO) test -p $(P) --race
-GOTESTNORACE := CGO_ENABLED=1 $(GO) test -p $(P)
-
+# Since TiDB add a new dependency on github.com/cloudfoundry/gosigar,
+# We need to add CGO_ENABLED=1 to make it work when build TiCDC in Darwin OS.
+# These logic is to check if the OS is Darwin, if so, add CGO_ENABLED=1.
+# ref: https://github.com/cloudfoundry/gosigar/issues/58#issuecomment-1150925711
+# ref: https://github.com/pingcap/tidb/pull/39526#issuecomment-1407952955
+OS    := "$(shell go env GOOS)"
 ARCH  := "$(shell uname -s)"
 LINUX := "Linux"
 MAC   := "Darwin"
+ifeq (${ARCH}, ${LINUX})
+	CGO := 0
```

Comment:
What if someone cross compile macOS TiCDC on a linux machine?

Reply:
I think only GOOS matters.

Reply:
Thank you!  You are right, only GOOS matters, but no matter set `CGO_ENABLED=0` or `CGO_ENABLED=1` run`GOOS=darwin make cdc` will always fails.
I revert related changes and run the following commands:
```shell
➜  ticdc git:(fix-get-source-id) uname
Linux
➜  ticdc git:(fix-get-source-id) uname -m
x86_64

➜  ticdc git:(fix-get-source-id) CGO_ENABLED=0 GOOS=darwin make cdc
GO111MODULE=on go build  -trimpath  -ldflags '-X "github.com/pingcap/tiflow/pkg/version.ReleaseVersion=v6.5.0-master-dirty" -X "github.com/pingcap/tiflow/pkg/version.BuildTS=2023-02-07 09:41:40" -X "github.com/pingcap/tiflow/pkg/version.GitHash=03e6f4650e98fe8b47788622844fc1a5f9272ead" -X "github.com/pingcap/tiflow/pkg/version.GitBranch=fix-get-source-id" -X "github.com/pingcap/tiflow/pkg/version.GoVersion=go version go1.19.3 linux/amd64"' -o bin/cdc ./cmd/cdc/main.go
# github.com/cloudfoundry/gosigar
vendor/github.com/cloudfoundry/gosigar/concrete_sigar.go:20:12: cpuUsage.Get undefined (type Cpu has no field or method Get)
vendor/github.com/cloudfoundry/gosigar/concrete_sigar.go:30:14: cpuUsage.Get undefined (type Cpu has no field or method Get)
vendor/github.com/cloudfoundry/gosigar/concrete_sigar.go:49:11: l.Get undefined (type LoadAverage has no field or method Get)
vendor/github.com/cloudfoundry/gosigar/concrete_sigar.go:55:11: m.Get undefined (type Mem has no field or method Get)
vendor/github.com/cloudfoundry/gosigar/concrete_sigar.go:61:11: m.GetIgnoringCGroups undefined (type Mem has no field or method GetIgnoringCGroups)
vendor/github.com/cloudfoundry/gosigar/concrete_sigar.go:67:11: s.Get undefined (type Swap has no field or method Get)
vendor/github.com/cloudfoundry/gosigar/sigar_shared.go:12:21: procTime.Get undefined (type *ProcTime has no field or method Get)
^Cmake: *** [cdc] Interrupt

➜  ticdc git:(fix-get-source-id) ✗ CGO_ENABLED=1 GOOS=darwin make cdc
GO111MODULE=on go build  -trimpath  -ldflags '-X "github.com/pingcap/tiflow/pkg/version.ReleaseVersion=v6.5.0-master-dirty" -X "github.com/pingcap/tiflow/pkg/version.BuildTS=2023-02-07 09:41:32" -X "github.com/pingcap/tiflow/pkg/version.GitHash=03e6f4650e98fe8b47788622844fc1a5f9272ead" -X "github.com/pingcap/tiflow/pkg/version.GitBranch=fix-get-source-id" -X "github.com/pingcap/tiflow/pkg/version.GoVersion=go version go1.19.3 linux/amd64"' -o bin/cdc ./cmd/cdc/main.go
# runtime/cgo
gcc: error: x86_64: No such file or directory
gcc: error: unrecognized command line option '-arch'
make: *** [cdc] Error 2
```
Do you have any idea to fix this problem?

Reply:
https://github.com/pingcap/tidb/pull/41165

use `CGO_ENABLED=1` is the only way to build on darwin.

Reply:
> [pingcap/tidb#41165](https://github.com/pingcap/tidb/pull/41165)
> 
> use `CGO_ENABLED=1` is the only way to build on darwin.

Yes, but can't build a darwin cdc on linux machine.